src/api/__tests__/Str.test.ts

```diff
@@ -0,0 +1,164 @@
+import CausalTree from '../CausalTree';
+
+describe('StrCursor', () => {
+  it('should insert a new char', () => {
+    const ct = new CausalTree();
+    const strCur = ct.setString().getCursor();
+
+    strCur.insert('a');
+    strCur.insert('b');
+    strCur.insert('c');
+    const newCh = strCur.insert('d');
+
+    expect(ct.toString()).toEqual(['abcd']);
+    expect(newCh.snapshot()).toEqual('d');
+  });
+  it('should return the pointed char', () => {
+    const ct = new CausalTree();
+    const strCur = ct.setString().getCursor();
+    strCur.insert('a');
+    strCur.insert('b');
+    strCur.insert('c');
+
+    const ch = strCur.element();
+
+    expect(ct.toString()).toEqual(['abc']);
+    expect(ch).toEqual('c');
+  });
+  it('should delete the pointed char and move the cursor backwards', () => {
+    const ct = new CausalTree();
+    const strCur = ct.setString().getCursor();
+    strCur.insert('a');
+    strCur.insert('b');
+    strCur.insert('c');
+
+    strCur.delete();
+
+    expect(ct.toString()).toEqual(['ab']);
+    expect(strCur.element()).toEqual('b');
+  });
+  it('should move the cursor accordingly', () => {
```

Comment:
Here, we only move 0->1->2, and then 2->(-1) is a special case. Please add a test where we create new cursors for every movement, and test 0->2, 0->3, and perhaps 0->3 passing over a deleted char.

Reply:
In this case, I primarily wanted to test the `index` method. So I'm not sure of what your suggestion is actually testing